pixeltable/catalog/view.py

```diff
@@ -284,21 +294,22 @@ def delete(self, where: Optional[exprs.Expr] = None) -> UpdateStatus:
         raise excs.Error(f'{self._display_str()}: Cannot delete from a {self._display_name()}.')
 
     def _get_base_table(self) -> Optional['Table']:
-        if self._tbl_version_path.base is None and not self._snapshot_only:
-            return None  # this can happen for a replica of a base table
-        # if this is a pure snapshot, our tbl_version_path only reflects the base (there is no TableVersion instance
-        # for the snapshot itself)
-        from pixeltable.catalog import Catalog
-
-        base_id = self._tbl_version_path.tbl_id if self._snapshot_only else self._tbl_version_path.base.tbl_id
-        with Catalog.get().begin_xact(tbl_id=base_id, for_write=False):
+        if self._tbl_version_path.tbl_id != self._id:
+            # _tbl_version_path represents a different schema object from this one. This can only happen if this is a
+            # named pure snapshot.
+            base_id = self._tbl_version_path.tbl_id
+        elif self._tbl_version_path.base is None:
+            return None
+        else:
+            base_id = self._tbl_version_path.base.tbl_id
+        with catalog.Catalog.get().begin_xact(tbl_id=base_id, for_write=False):
             return catalog.Catalog.get().get_table_by_id(base_id)
 
     @property
     def _effective_base_versions(self) -> list[Optional[int]]:
         effective_versions = [tv.effective_version for tv in self._tbl_version_path.get_tbl_versions()]
-        if self._snapshot_only:
-            return effective_versions
+        if self._snapshot_only and not self._is_anonymous_snapshot():
```

Comment:
Can't we treat an anonymous snapshot as a pure snapshot (and then not special-case it here and elsewhere)?